src/elliptic.jl

```diff
@@ -0,0 +1,245 @@
+# References:
+#  [1] Abramowitz, Stegun: Handbook of Mathematical Functions (1965)
+#  [2] ellipjc.m in Toby Driscoll's Schwarz-Christoffel Toolbox
+
+
+#------------------------------------------------
+# Descending and Ascending Landen Transformation
+
+descstep(m) = m/(1+sqrt(1-m))^2
+
+@generated function shrinkm(m,::Val{N}) where {N}
+    # [1, Sec 16.12] / https://dlmf.nist.gov/22.7.i
+    quote
+        f = one(m)
+        Base.Cartesian.@nexprs $N i->begin
+            k_i = descstep(m)
+            m = k_i^2
+            f *= 1+k_i
+        end
+        return (Base.Cartesian.@ntuple $N k), f, m
+    end
+end
+
+function ellipj_smallm(u,m)
+    # [1, Sec 16.13] / https://dlmf.nist.gov/22.10.ii
+    if VERSION < v"0.7-"
+        sinu = sin(u)
+        cosu = cos(u)
+    else
+        sinu, cosu = sincos(u)
+    end
+    sn = sinu - m*(u-sinu*cosu)*cosu/4
+    cn = cosu + m*(u-sinu*cosu)*sinu/4
+    dn = 1 - m*sinu^2/2;
+    return sn,cn,dn
+end
+function ellipj_largem(u,m1)
+    # [1, Sec 16.15] / https://dlmf.nist.gov/22.10.ii
+    sinhu = sinh(u)
+    coshu = cosh(u)
+    tanhu = tanh(u)
+    sechu = sech(u)
+    sn = tanhu + m1*(sinhu*coshu-u)*sechu^2/4
+    cn = sechu - m1*(sinhu*coshu-u)*tanhu*sechu/4
+    dn = sechu + m1*(sinhu*coshu+u)*tanhu*sechu/4
+    return sn,cn,dn
+end
+
+function ellipj_growm(sn,cn,dn, k)
```

Comment:
these could just be combined into the `ellipj_via*` functions.

Reply:
I could, but
- I find it more readable if the coarse structure of the algorithm is separated from the details,
- it is easier to debug if you can prod the different parts separately (I often found it convenient to check whether a numerical instability originated in `_smallm`/`_largem` or `_growm`/`_shrinkm`), and
- there might be some expert users who would want to evaluate the limit formulas without the Landen steps, and like this they can.